ForgeRock/forgerock-web-login-framework
# fix: addresses bugs reported in beta testing and doc changes for new api

## Summary

This PR addresses feedback from beta testing of the Login Widget. This fixes the following:

1. Errors within journey flow were not being reported. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-8dcfd178b5d5057e0274c71b248321bc4d51d86009561d615a8a0a92a6d4ed3bR180
2. The configuration setting `timeout` is now documented as required. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R262 & https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R495
3. Although not recommended, setting `tree` on the `forgerock` config property now works. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-8dcfd178b5d5057e0274c71b248321bc4d51d86009561d615a8a0a92a6d4ed3bR177
4. The `links` property has better documentation. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R760
5. Terms and conditions component now follows style `checksAndRadios` config. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-e5c63b92e5f3e948839652e44daae0340652510a19f5bbc4d56c24fb46f01c04L37
6. Security questions are now required by default. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-711981177cd829fc3bd747b69c55c7ed31a3fae01d637d417b1da0abdf9f5270R185
7. Labels for attribute collectors now follows `labels` style config. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-d0c93bcf04db462096faf2fd6051d2487a3be75b97fb23c7d7716c012dc15e38R36
8. `NumberAttributeInputCallback` is added to not supported list. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R805
9. `ChoiceCallback` component handler now follows `checksAndRadios` style config. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-0a52584ed0a88df071bcf41eea1603987b15ee94194b3f6ffef817973e1c21f8R27
10. `ChoiceCallback` component handler now follows `labels` style config. Fix: https://github.com/ForgeRock/forgerock-web-login-framework/pull/75/files#diff-0a52584ed0a88df071bcf41eea1603987b15ee94194b3f6ffef817973e1c21f8R28


## How to verify

For most of the fixes related to the style config, you can go to visit the latest Storybook from this PR: https://64075ad87b9e160633c33298-tohrhqdxhf.chromatic.com/?path=/story/style-config--defaults. I've added a new section that relates to verifying the use of the style config settings and an example registration journey (see image).

![CleanShot 2023-03-14 at 16 04 36](https://user-images.githubusercontent.com/3070458/225136734-46c05341-29ca-4426-a6a4-2ba6686cb685.png)

To test the journey flow error handling, you'll have to pull this in locally and alter the config object you pass into the `config.set` method.
